InvenTree/part/serializers.py

```diff
@@ -437,6 +440,31 @@ class Meta:
             'virtual',
         ]
 
+    def save(self):
+        """Save the Part instance"""
+
+        super().save()
+
+        part = self.instance
+
+        # Check if an image was downloaded from a remote URL
+        remote_img = getattr(self, 'remote_image_file', None)
+
+        if remote_img and part:
+            fmt = remote_img.format or 'PNG'
+            buffer = io.BytesIO()
+            remote_img.save(buffer, format=fmt)
+
+            # Construct a simplified name for the image
+            filename = f"part_{part.pk}_image.{fmt.lower()}"
+
+            part.image.save(
+                filename,
+                ContentFile(buffer.getvalue()),
+            )
```

Comment:
Suggestion: This could be a general override in the `RemoteImageMixin` class - would reduce code.